lumicks/pylake/population/dwelltime.py

```diff
@@ -544,8 +584,96 @@ def exponential_mixture_log_likelihood(params, t, min_observation_time, max_obse
     return -np.sum(log_likelihood)
 
 
+def _handle_amplitude_constraint(
+    n_components, params, fixed_param_mask
+) -> Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]:
+    """Enforce amplitude constraints
+
+    Determines how many amplitudes actually need fitting. If only a single amplitude needs
+    "fitting" then the result is already pre-determined by the problem definition and this function
+    will return a parameter vector with the parameter updated to its correct value.
```

Comment:
Single description sentence first is part of the numpy [style](https://numpydoc.readthedocs.io/en/latest/format.html#short-summary) though. I do like the suggestion though.

The description is not entirely what the function does in its present state (since we still fit two amplitudes in the two-component case). How about:
```
Determines how many amplitudes actually need fitting.
    
For a single-component model the amplitude is 1 by definition. For an N-component model, where
N - 1 amplitudes are fixed, the free amplitude is determined by the constraint. For models that
have more than one free amplitude, the amplitudes are constrained to sum to 1.
```

Reply:
👍 ok, I remember the idea for that had come up in discussions, but indeed seems like a lot of bloat for little payoff